### ISSUE TITLE
Wrong type reported when type is correct

### ISSUE THREAD

**Issue by Rizhiy** [microsoft/pyright#8020]
**Describe the bug**
In the following code:
```python
from _thread import LockType
from threading import Lock

lock: LockType = Lock()
```
I get the following error:
```
error: Expression of type "Lock" is incompatible with declared type "LockType"
    "Lock" is incompatible with "LockType" (reportAssignmentType)
```
But `Lock` does return `LockType` (it is actually a function), e.g. for me (python 3.11):
```python
# python3.11/threading.py
_allocate_lock = _thread.allocate_lock
Lock = _allocate_lock
# python3.11
```
https://github.com/microsoft/pyright/blob/2e397e9a0a84818342712af7639114b48e274a7a/packages/pyright-internal/typeshed-fallback/stdlib/_thread.pyi#L30

**VS Code extension or command-line**
Command line:
```
Python 3.11.7
pyright 1.1.364
```

**Comment by erictraut** [comment]
Pyright is working correctly here, so I don't consider this a bug.

Pyright, like other type checkers such as mypy, bases its type analysis on type information provided in type stub files provided by the typeshed project. You can see how the stubs define the `Lock` class [here](https://github.com/python/typeshed/blob/c82d29fc7614a9ecd33bd416386062ba2915850d/stdlib/threading.pyi#L104).

If you think this type information is incorrect, please file a bug with the typeshed project.
